### ISSUE TITLE
Local iis host port node error EACCESS

### ISSUE THREAD

**Issue by DavidWhit** [projectkudu/kudu#2204]
I realize this is a node error though all the tutorials and information show this should work in 
configuring IIS with node just like the standard iis node.

When I host on port 80 and attempt to listen on port 80 I get eacces uncaught exception.
The application has local system permissions and the permissions are not read only. Can anyone clue
me in on what is missing?

Application has thrown an uncaught exception and is terminated:
Error: listen EACCES "inhouseip":80
    at Object.exports._errnoException (util.js:1026:11)
    at exports._exceptionWithHostPort (util.js:1049:20)
    at Server._listen2 (net.js:1240:19)
    at listen (net.js:1289:10)
    at net.js:1399:9
    at _combinedTickCallback (internal/process/next_tick.js:77:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
    at Module.runMain (module.js:592:11)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)

**Comment by davidebbo** [comment]
When using iisnode, instead of listening on 80, you need to listened on port `process.env.PORT`.

**Comment by DavidWhit** [comment]
Alright, so I tried setting the port environment variable to 80 and restarted the kudu iis application and have the same result. I also removed the PORT environment var but of course it's undefined at that point so the result was erroneous. 

I went with a single basic example so I could find the error easier 
webconfig = 
```
<configuration>
   <system.webServer>
     <handlers>
       <add name="iisnode" path="server.js" verb="*" modules="iisnode" />
     </handlers>
     <rewrite>
       <rules>
         <rule name="basic">
           <match url="/*" />
           <action type="Rewrite" url="server.js" />
         </rule>
       </rules>
     </rewrite>
     <security>
       <requestFiltering>
         <hiddenSegments>
           <add segment="node_modules" />
         </hiddenSegments>
       </requestFiltering>
     </security>    
   </system.webServer>
 </configuration>
```

server.js = 
```
var app = require('express')();

 app.get('/basic', function (req, res) {
     res.send('test');
 });

app.listen(process.env.PORT, 'internalip');
```

**Comment by davidebbo** [comment]
It should be automatically defined when running on iisnode (which Azure Web Apps do), so you don't need to set it. It will not be 80.

**Comment by DavidWhit** [comment]
Thanks Dave and all your help with this unsupported setup.

I never had much to go on other than something wasnt right. I did learn
that iis node sets the port so I did remove that and refreshed the iis
server and still had the same issue. I resetup from scratch after being to
frustrated to continue when everything seems to be as other examples show
even the basic of examples. Once I methodically did that everything is
working as expected. At least I came out learning more than I did when I
started.

Thanks again
